src/dysh/spectra/spectrum.py

```diff
@@ -926,6 +932,73 @@ def _div_meta(self, operand, operand2=None, **kwargs):
         # TBD
         return deepcopy(operand)
 
+    def __getitem__(self, item):
+
+        def q2idx(q, wcs, spectral_axis, coo, sto):
+            """Quantity to index."""
+            if "velocity" in u.get_physical_type(q):
+                # Convert from velocity to channel.
+                idx = vel2idx(q, wcs, spectral_axis, coo, sto)
+            elif "frequency" in u.get_physical_type(q):
+                # Convert from frequency to channel.
+                idxs = wcs.world_to_pixel(coo, q, sto)
+                idx = int(np.round(idxs[0]))
+            return idx
+
+        def vel2idx(vel, wcs, spectral_axis, coo, sto):
+            eq = get_spectral_equivalency(spectral_axis.doppler_rest, 
+                                          spectral_axis.doppler_convention)
+            # Make `vel` a `SpectralCoord`.
+            vel_sp = spectral_axis.to(unit=vel.unit).replicate(value=vel.value, unit=vel.unit)
```

Comment:
should this line be inside the `with` clause?  In wav2idx, you have it inside the `with`.
also wav2idx and vel2idx are very similar, could be combined into a single method with common signature, e.g.

```
def get_idx(q, wcs,spectral_axis, coo, sto, eq):
    with u.set_enabled_equivalencies(eq):
          sp = spectral_axis.to(q.unit).replicate(value=q.value,unit=q.unit)
          idxs = wcs.world_to_pixel(coo,sp,sto)
    return int(np.round(idxs[0]))
```
and calculate `eq` in q2idx before calling get_idx

Reply:
When I try to transform the `spectral_axis` inside the context manager I get the following error:
```Python
ValueError: 'km/s' did not parse as unit: maximum recursion depth exceeded If this is meant to be a custom unit, define it with 'u.def_unit'. To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. For details, see https://docs.astropy.org/en/latest/units/combining_and_defining.html
```
(The link is not very helpful.) 

That is why the transformation takes place outside the context manager, and why there are 2 separate methods. If you know how to bypass the error, we could try again.

Reply:
ok, let's not worry about refactoring this small bit of code